### PR TITLE
[LMS-360][Integration] User List Sorting Column

### PR DESCRIPTION
**Pull request by ragg09-sunasterisk**
## Issue Link
[LMS-360 [Integration] Sorting Column](https://framgiaph.backlog.com/view/LMS-360)

## Defintion of Done
- Update User Serializer
-- Role return ID and Title
-- Status return ID and Name
- can click either first_name. last_name, and email and sort the data by that attribute
- down | up button changing on click
- show arrow only in the selected header

## Steps to reproduce
_will update soon..._

## Affected Components / Functionalities / Page
n/a

## Test Cases
_will update soon..._

### Todo
- FE/BE integration
- code clean up
- rebase

### other
- I am just logging the data for now, but you can see in that log that everytime header is clicked, there is a request and proper sorting

## Screenshots
sorting

### CODE REVIEW

File: client/src/shared/hooks/useShowUserList.ts [hunk 1]
```diff
@@ -63,19 +78,48 @@ const useShowUserList = (): any => {
...
+            newSortOrder ? 'asc' : 'desc'
+          }&page_size=${limiter}`
+        );
+        if (isRequestOk(result)) {
+          setShowPerPage(result.data.user);
+          setCurrentPage(1);
+        }
+      } catch (error) {
+        console.error(error);
+        alert('something went wrong');
```
Comment (mikorosales) [inline client/src/shared/hooks/useShowUserList.ts:102]: toastify alert instead of native :)

**Review by mikorosales — APPROVED** [review]

**Comment by Marielleregner7** [comment]
Comment:
- The arrows beside each column header are not pointing to the correct way when sorting. 

↑ = ascending
↓ = descending
ascdec

**Comment by Marielleregner7** [comment]
Comment:
- Encountered an error when running` npm run build`, please check  🙇 

image

**Comment by ragg09-sunasterisk** [comment]
image

**Comment by Marielleregner7** [comment]
Comment:
- There is another one sir  🙇 
- 
image

**Review by Marielleregner7 — APPROVED** [review]
**APPROVED**
- [ ] Able to sort (asc, desc) first_name. last_name, and email.
- [x] Can click Up arrow for ascending and down arrow for descending order.
- [x] No ESLint error.

Comments: 6
Test time: 45 mins
Iterations: 3

Proof:
sortinguserlist
image

**Pull request merged**